tutor/src/components/course-listing/listings.cjsx

```diff
@@ -0,0 +1,155 @@
+_ = require 'lodash'
+React = require 'react'
+BS = require 'react-bootstrap'
+{Link} = require 'react-router'
+classnames = require 'classnames'
+
+Router = require '../../helpers/router'
+
+{CourseStore} = require '../../flux/course'
+{CurrentUserStore} = require '../../flux/current-user'
+
+CourseData = require '../course-data-mixin'
+IconAdd = require  '../icons/add'
+
+{Course, CourseTeacher} = require './course'
+
+getReactBaseName = (context) -> _.kebabCase(context.constructor.displayName)
+
+wrapCourseItem = (Item, course = {}) ->
+  {id} = course
+  courseName = id or 'new'
+
+  if id
+    courseDataProps = CourseData.getCourseDataProps(id)
+    courseSubject = CourseStore.getSubject(id)
+    courseIsTeacher = CourseStore.isTeacher(id)
+
+  <BS.Col key="course-listing-item-wrapper-#{courseName}" md={3} sm={6} xs={12}>
+    <Item
+      course={course}
+      courseSubject={courseSubject}
+      courseIsTeacher={courseIsTeacher}
+      courseDataProps={courseDataProps}/>
+  </BS.Col>
+
+DEFAULT_COURSE_ITEMS =
+  teacher: CourseTeacher
+  student: Course
+
+CourseListingBase = React.createClass
+  displayName: 'CourseListingBase'
+
+  getItems: ->
+    _.merge({}, DEFAULT_COURSE_ITEMS, @props.Items)
+
+  render: ->
+    {courses, className, before, after} = @props
+    Items = @getItems()
+
+    sectionClasses = classnames('course-listing-section', className)
+
+    <BS.Row className={sectionClasses}>
+      {before}
+      {_.map(courses, (course) ->
+        Item = Items[CurrentUserStore.getCourseRole(course.id)]
+        if Item then wrapCourseItem(Item, course)
+      )}
+      {after}
+    </BS.Row>
+
+AddCourseArea = ->
+  <Link
+    to={Router.makePathname('createNewCourse')}
+    className='course-listing-add-zone'
+  >
+    <div>
+      <IconAdd/>
+      Add a course
+    </div>
+  </Link>
+
+CourseListingCurrent = React.createClass
+  displayName: 'CourseListingCurrent'
+  NoCourses: ->
+    <BS.Row className='course-listing-none'>
+      <BS.Col md={12}>
+        <p>There are no current courses.</p>
+      </BS.Col>
+    </BS.Row>
+
+  Title: ->
+    baseName = getReactBaseName(@)
+
+    <BS.Row className='course-listing-title'>
+      <BS.Col md={12}>
+        <h1>Current Courses</h1>
+      </BS.Col>
+    </BS.Row>
+
```

Comment:
`baseName` doesn't appear to be used here.   

Since the `NoCourses` and `Title` methods aren't accessing any props they could be created outside the `CourseListingCurrent` class as standalone functions.  That'd be a smidge better since then they'd be more isolated and not have to go through React's `this` binding layer when called.